Cython/Utility/ModuleSetupCode.c

```diff
@@ -1588,15 +1588,15 @@ static int __Pyx_check_binary_version(void);
 /////////////// CheckBinaryVersion ///////////////
 
 static int __Pyx_check_binary_version(void) {
-    char ctversion[4], rtversion[4];
-    PyOS_snprintf(ctversion, 4, "%d.%d", PY_MAJOR_VERSION, PY_MINOR_VERSION);
-    PyOS_snprintf(rtversion, 4, "%s", Py_GetVersion());
-    if (ctversion[0] != rtversion[0] || ctversion[2] != rtversion[2]) {
+    int ct_major_version = PY_MAJOR_VERSION, ct_minor_version = PY_MINOR_VERSION;
+    int rt_major_version, rt_minor_version;
+    sscanf(Py_GetVersion(), "%d.%d", &rt_major_version, &rt_minor_version);
```

Comment:
I can't find any other uses of `scanf`/`sscanf` or similar in Cython, which makes me think we might be deliberately avoiding them. But it does seem the simplest solution here.